packages/xrpc/src/util.ts

```diff
@@ -85,100 +111,343 @@ export function encodeQueryParam(
+        ? data.constructor.name
+        : 'object'
+
+  throw new XRPCError(
+    ResponseType.InvalidRequest,
+    `Unable to encode ${type} as ${contentType} data`,
+  )
 }
 
-export function httpResponseCodeToEnum(status: number): ResponseType {
-  let resCode: ResponseType
-  if (status in ResponseType) {
-    resCode = status
-  } else if (status >= 100 && status < 200) {
-    resCode = ResponseType.XRPCNotSupported
-  } else if (status >= 200 && status < 300) {
-    resCode = ResponseType.Success
-  } else if (status >= 300 && status < 400) {
-    resCode = ResponseType.XRPCNotSupported
-  } else if (status >= 400 && status < 500) {
-    resCode = ResponseType.InvalidRequest
-  } else {
-    resCode = ResponseType.InternalServerError
-  }
-  return resCode
+/**
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/from_static}
+ */
+function iterableToReadableStream(
+  iterable: Iterable<unknown> | AsyncIterable<unknown>,
+): ReadableStream<Uint8Array> {
+  // Use the native ReadableStream.from() if available.
+  if ('from' in ReadableStream && typeof ReadableStream.from === 'function') {
+    return ReadableStream.from(iterable)
+  }
+
+  // Note, in environments where ReadableStream is not available either, we
+  // *could* load the iterable into memory and create an Arraybuffer from it.
+  // However, this would be a bad idea for large iterables. In order to keep
+  // things simple, we'll just allow the anonymous ReadableStream constructor
+  // to throw an error in those environments, hinting the user of the lib to find
+  // an alternate solution in that case (e.g. use a Blob if available).
+
+  let generator: AsyncGenerator<unknown, void, undefined>
+  return new ReadableStream<Uint8Array>({
+    type: 'bytes',
+    start() {
+      // Wrap the iterable in an async generator to handle both sync and async
+      // iterables, and make sure that the return() method exists.
+      generator = (async function* () {
+        yield* iterable
+      })()
+    },
+    async pull(controller: ReadableStreamDefaultController) {
+      const { done, value } = await generator.next()
+      if (done) {
+        controller.close()
+      } else {
+        try {
+          const buf = toUint8Array(value)
+          if (buf) controller.enqueue(buf)
+        } catch (cause) {
+          // ReadableStream won't call cancel() if the stream is errored.
+          await generator.return()
+
+          controller.error(
+            new TypeError(
+              'Converting iterable body to ReadableStream requires Buffer, ArrayBuffer or string values',
+              { cause },
+            ),
+          )
+        }
+      }
+    },
+    async cancel() {
+      await generator.return()
+    },
+  })
 }
 
+// Browsers don't have Buffer. This syntax is to avoid bundlers from including
+// a Buffer polyfill in the bundle if it's not used elsewhere.
+const globalName = `${{ toString: () => 'Buf' }}fer` as 'Buffer'
```

Comment:
This is giving some build tools some issues.  Next.js projects choke on it during build, for example, here: https://github.com/vercel/nft/blob/a930bc78911aa1af6326382038730461ae828a8b/src/utils/static-eval.ts#L525
```
TypeError: Cannot convert object to primitive value
```


Is this common bundler functionality that we indeed do need to circumvent?  If so, is there another way we could do it that doesn't cause issues for Next.js?

Reply:
This purpose of this is to support using an iterable as XRPC data on most environments, by providing a polyfill for `ReadableStream.from`, when `ReadableStream` is available.

I've remove this code and replaced it with a comment recommending the user of the lib to provide their own polyfill for `ReadableStream.from` if they need to support using iterables as XRPC request body.

https://github.com/bluesky-social/atproto/pull/2483/commits/3681c907b29762e2c4ab965894d02a67484b85e5

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/from_static

Note that chrome does not support `ReadableStream.from`:
https://issues.chromium.org/issues/40267032

Reply:
Coolcool!  Not sure if it makes a difference, but it was specifically this line where I bumped into the issue:
```ts
const globalName = `${{ toString: () => 'Buf' }}fer` as 'Buffer'
```

The build tool was trying to do some static evaluation and didn't know what to do with `{ toString: () => 'Buf' }` within the tagged template literal.

Reply:
Which was my exact intent when writing this XD

I expected that bundlers would just ignore unexpected patterns instead of throwing. My assumption being wrong, I just removed this.